### ISSUE TITLE
Heavy stuttering whilst using Godot editor on a G Sync enabled monitor

### ISSUE THREAD

**Issue by MichaelSavaglia** [godotengine/godot#96125]
### Tested versions

4.3 stable
4.3 dev

### System information

Windows 11, Godot 4.3 stable, Vulkan Forward+

### Issue description

The editor stutters frequently whilst using the editor. It's most easily reproducible whilst highlighting text in the text editor
disabling G Sync on the monitor resolves the issue.
Monitor used is ROG SWIFT PG278QR - I have not been able to test on any other G Sync monitors, but I tested on another non G sync monitor and there was no issue

### Steps to reproduce

Enable G Sync on monitor.
Open Editor
Highlight text in the text editor
The PC will now stutter for a second or two.

### Minimal reproduction project (MRP)

I have been able to reproduce this in any Godot project, including the small samples and empty projects

**Comment by MichaelSavaglia** [comment]
Enabling "Update Continuously" in the editor settings is an alternative fix

**Comment by Arondy** [comment]
Do you have G-Sync enabled in windowed and fullscreen mode in Nvidia Control Panel? Maybe you should consider switching it to fullscreen only.

**Comment by MichaelSavaglia** [comment]
G-Sync is only enabled for full screen.
Do you have Update Continuously selected in your settings? If so try disabling that when attempting a repro, if you haven't already

**Comment by Arondy** [comment]
Nope, it is turned off and was so when i tried to reproduce the issue.

**Comment by Isabello** [comment]
I had this problem a few months ago and I solved it by disabling GSync when using Godot. 

I tried re-enabling Gsync to help try and reproduce here. I tested both fullscreen and fullscreen windowed options and there are no issues on stable or the older beta versions of 4.3 or even 4.2.1 which is the version I originally had this issue on. I am running Windows 10 so its not a windows version issue, you may need to take a driver update.

**Comment by Edgexx** [comment]
I'm also experience the same issue. Disabling G-Sync via NVIDIA Control Panel resolves the issue.
My monitor: Acer Predator XB271HU
I was using G-Sync "Enabled for windowed and full screen mode"
Godot_v4.3-stable_mono_win64.exe

**Comment by magian1127** [comment]
win11 NVIDIA 561.09 godot4.3+ 
Simply enabling fullscreen G-Sync doesn't work either. have to disable G-Sync entirely. 
This issue only occurs in Godot under 'Forward+'.

**Comment by mangiavellano** [comment]
Exact same issue here, I've tried to disable the "Vsync" option inside the project settings but it doesn't change anything.
Even when I run the game in windowed mode, the entire screen is stuttering (even things outside of the game's window).

The only way to avoid the stuttering is to completely disable the G-Sync option in the Nvidia settings.

I'm working on macOS (Godot 4.3.stable, Mobile Rendering) but the issue happens when exporting to Windows and running the game on the following setup:

Win11, NVIDIA GeForce GTX 1660 SUPER, NVIDIA 566.36, AOC Gaming 27G2G8 monitor 244Hz

**Comment by yurirb9** [comment]
This issue is happening with me as well under the same conditions as mentioned above.

**Comment by Calinou** [comment]
Windowed G-Sync is supposed to be disabled since 4.3 thanks to https://github.com/godotengine/godot/pull/93737. I guess the profile isn't working for some reason?

Can anyone try to use [NVIDIA Profile Inspector](https://github.com/Orbmu2k/nvidiaProfileInspector) and try to override the G-Sync state for the Godot editor executable, and see if it resolves the issue without needing to change any editor setting?

cc @aitorcki

**Comment by Fexty12573** [comment]
> Can anyone try to use [NVIDIA Profile Inspector](https://github.com/Orbmu2k/nvidiaProfileInspector) and try to override the G-Sync state for the Godot editor executable, and see if it resolves the issue without needing to change any editor setting?

This does not solve the issue on my PC. I have tried setting the G-Sync Application Mode to both "Fullscreen Only" as well as "Off" and neither option gets rid of the stuttering. Only disabling G-Sync globally via the NVIDIA Control Panel seems to affect anything.

For reference I am using an NVIDIA RTX 5080 with driver version 576.40 and my display is an ASUS ROG SWIFT PG279QE.

**Comment by kurushimee** [comment]
In my case, *sometimes* the editor experiences intense VRR flicker (I can see my refresh rate going all over the place, from 60 to 280 and everything in-between). I have no idea why G-Sync tries to work with the Godot Editor, when it's a windowed application, and G-Sync is set to only work for fullscreen applications?

Nevertheless, what fixes this is getting NVIDIA Profile Inspector and editing the profile for the Godot Editor to change both `GSYNC - Application Requested State` and `GSYNC - Application State` to `Disallow`.

If you try to set "monitor technology" for Godot Editor to "Fixed Refresh", it works too, but then the monitor turns to black whenever you switch applications, because it fully turns off G-Sync when in the Godot Editor, instead of just not using it.

**Comment by Calinou** [comment]
> Nevertheless, what fixes this is getting NVIDIA Profile Inspector and editing the profile for the Godot Editor to change both `GSYNC - Application Requested State` and `GSYNC - Application State to Disallow`.

cc @aitorciki 

It was mentioned a while ago that we should have a dedicated application profile for the Godot editor: https://github.com/godotengine/godot/pull/93737#issuecomment-2203154973

It could probably set the options mentioned here. The downside is that VRR would no longer work on a fullscreen Godot editor or running project (until you export it, which will make it use a different profile).

That said, I haven't been able to reproduce this issue on Windows 11 24H2 and a GeForce RTX 5090 with the latest drivers (using an ASUS ROG PG32UCDP). I have G-Sync enabled for fullscreen applications only.

**Comment by mangiavellano** [comment]
@Calinou I've run more tries, the issue is still here with the latest Nvidia driver and Godot 4.4.1. (Win11, NVIDIA GeForce GTX 1660 SUPER). Still need to fully disable G-Sync.

The Godot rendering mode doesn't matter (I've tried all of them, same result)

**A clue though**, the issue seems **to happen only on my 240hz screen**, when I try with the exact same configuration but on a 144hz monitor, everything seems fine (in the editor and in the exported project).

I've checked all the monitors specs listed in this thread and it seems that all of them are above 144hz, could be a lead?

**Comment by Calinou** [comment]
> I've checked all the monitors specs listed in this thread and it seems that all of them are above 144hz, could be a lead?

In this case, the issue is that the editor is capped to 145 FPS by default due to the **Low Processor Mode Sleep Usec** Editor Setting that defaults to `6900`. (The maximum FPS that can be reached is determined by `1000000.0/6900`.)

If you change it to `4100` on 240 Hz and restart the editor, the issue *should* disappear, at least if the CPU/GPU can keep up.

We should revise this editor setting to use a value that matches the fastest's monitor refresh rate by default, so that it can fully make use of high refresh rate monitors out of the box. This probably involves adding an enum editor setting to choose between an automatic value and the custom existing value (and changing this custom value to be expressed in FPS rather than µspf, as it's more user-friendly).

I also have V-Sync disabled in the Editor Settings here; see if that helps too. Enabling Single Window Mode should further help reduce lag (as new windows won't spawn anymore and [compete with V-Sync rates](https://github.com/godotengine/godot/issues/93014#issuecomment-2197799816)), but it will disable some functionality in the editor.

**Comment by Calinou** [comment]
I've implemented something to the effect of https://github.com/godotengine/godot/issues/96125#issuecomment-3693575065:

- https://github.com/godotengine/godot/pull/115548

@havidarou Double-check that G-Sync is only enabled on fullscreen applications, not fullscreen and windowed applications in the NVIDIA Control Panel.

**Comment by kurushimee** [comment]
> Double-check that G-Sync is only enabled on fullscreen applications, not fullscreen and windowed applications in the NVIDIA Control Panel.

Yeah, this G-Sync option is misleading/plain not working correctly - Godot is not the only app that somehow enables G-Sync in windowed mode even when G-Sync is strictly set to Fullscreen-only

**Comment by EMBYRDEV** [comment]
I am also facing this, this is a fairly serious issue for anyone making applications as opposed to games with Godot. Low Processor Mode is basically unusable.

**Comment by omltcat** [comment]
> Nevertheless, what fixes this is getting NVIDIA Profile Inspector and editing the profile for the Godot Editor to change both `GSYNC - Application Requested State` and `GSYNC - Application State` to `Disallow`.

Can confirm this indeed works. After disabling these options, Godot editor is mostly stutter free. An alternative workaround is to use single window mode. Hopefully there will be an official solution soon.
